### ISSUE TITLE
$member->Groups()->removeAll() doesn’t work

### ISSUE THREAD

**Issue by kinglozzer** [silverstripe/silverstripe-framework#3948]
`Couldn't run query: DELETE FROM "Group_Members" WHERE ("Group"."ID" IN (3)) AND ("Group_Members"."GroupID" IN SELECT "Group"."ID" FROM "Group")) Unknown column 'Group.ID' in 'where clause'`

It should be looking for `Group_Members.GroupID` instead of `Group.ID` in the WHERE clause, but the table/column are hard-coded here: https://github.com/silverstripe/silverstripe-framework/blob/773624a40b2b87283c6b75da2b313caa2e6f683f/security/Member.php#L1646.

**Comment by kinglozzer** [comment]
Okay so `$member->DirectGroups()->removeAll()` works as that just uses `ManyManyList`, whereas `Groups()` includes parent groups too. I’d still like to see this fixed though, as the behaviour of removing all groups should be consistent across both lists.

**Comment by dhensby** [comment]
POSSIBLY related to #3839?

**Comment by ShiftedBit** [comment]
```
$groups = $member->Groups();

foreach($groups as $group) {
    $member->Groups()->remove($group);
}
```
that works for me. in the meantime.

**Comment by dhensby** [comment]
yes, the problem is `Groups` is a function defined here: https://github.com/silverstripe/silverstripe-framework/blob/773624a40b2b87283c6b75da2b313caa2e6f683f/security/Member.php#L1147

and to acess the relation you have to use `DirectGroups()` this is a broken expectation but not actually unintended.

**Comment by tractorcow** [comment]
I've closed #1681 in favour of this ticket.

**Comment by tractorcow** [comment]
Fixed with #7798
